lampod/src/chain/blockchain.rs

```diff
@@ -27,13 +37,59 @@ impl LampoChainManager {
         LampoChainManager {
             backend: client,
             wallet_manager,
+            current_height: Arc::new(Mutex::new(Height::from_consensus(0).unwrap())),
+            best_height: Arc::new(Mutex::new(Height::from_consensus(0).unwrap())),
+            handler: RefCell::new(None),
         }
     }
 
     pub fn is_lightway(&self) -> bool {
         self.backend.is_lightway()
     }
 
+    pub fn set_handler(&self, handler: Arc<LampoHandler>) {
+        self.handler.replace(Some(handler));
+    }
+
+    pub fn handler(&self) -> Arc<LampoHandler> {
+        self.handler.borrow().clone().unwrap()
+    }
+
+    pub fn is_syncing(&self) -> bool {
+        const SYNC_THRESHOLD: u32 = 1;
+        let current_height = self.current_height.lock().unwrap();
+        let best_height = self.best_height.lock().unwrap();
+        current_height.to_consensus_u32() + SYNC_THRESHOLD < best_height.to_consensus_u32()
+    }
+
+    pub fn listen(&self) -> error::Result<()> {
+        let _backend_listner = self.backend.clone().listen()?;
+        let event_listener = self.clone();
+        std::thread::spawn(move || {
+            log::info!(target: "lampo_chain_manager", "listening for chain events");
+            loop {
+                let handler = event_listener.handler();
+                let event = handler.events().recv().unwrap();
+                match event {
+                    Event::OnChain(OnChainEvent::NewBlock(block)) => {
```

Comment:
Hey @vincenzopalazzo 
Need your help here
I did not see any `NewBlock` event emitted in the logs, only `NewBestBlock` was emitted

Reply:
Mh idk we should debug the backend, I am pretty comfortable that I made some mistake there.